win32ss/user/user32/misc/exticon.c

```diff
@@ -714,9 +714,75 @@ static UINT ICO_ExtractIconExW(
 	    xresdir = find_entry_by_id(iconresdir, LOWORD(pIconId[i]), rootresdir);
             if( !xresdir )
             {
+#ifdef __REACTOS__
+              const IMAGE_RESOURCE_DIRECTORY *resdir;
+              WARN("icon entry %d not found\n", LOWORD(pIconId[i]));
+
+              /* Try to get an icon by walking the files Resource Section */
+              if (iconresdir->NumberOfIdEntries > 0)
+              {
+                  xresent = (const IMAGE_RESOURCE_DIRECTORY_ENTRY*)
+                      (ULONG_PTR)(iconresdir + 1);
+              }
+              else
+              {
+                  RetPtr[i] = 0;
+                  continue;
```

Comment:
Create/Use a `goto`, instead of duplicating existing code.

Reply:
No thanks. Maybe you can do another PR later for this.

Reply:
Hum :-|